### PR TITLE
Isolate state building stages in `run_test_case`

### LINKED ISSUE #809: Isolate state building stages in `running.rs`

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**
Closes #809 

## Introduced changes

- Splits run test case into smaller stages

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Comment by MaksymilianDemitraszek** [comment]
It isolates building stages and separates data as state, adds `TestExecutionState`. Further plan is to separate state building and test running.

**Review by Arcticae — APPROVED** [review]

File: crates/cheatnet/src/execution/syscalls.rs [hunk 1]
```diff
@@ -118,6 +118,18 @@ pub struct CheatableSyscallHandler<'a> {
     pub cheatnet_state: &'a mut CheatnetState,
 }
 
+impl<'a> CheatableSyscallHandler<'a> {
+    pub fn from(
```
Comment (piotmag769) [inline crates/cheatnet/src/execution/syscalls.rs:122]: Looks more like `new` to me

File: crates/forge/src/test_execution_syscall_handler.rs [hunk 2]
```diff
@@ -60,15 +60,34 @@ impl From<&StarknetContractArtifacts> for ContractArtifacts {
...
+    pub test_execution_state: &'a mut TestExecutionState<'a>,
+    // we need to keep a copy of hints as SyscallHintProcessor keeps it as private
     pub hints: &'a HashMap<String, Hint>,
-    // pub cheatnet_state: CheatnetState,
     pub run_resources: RunResources,
-    pub environment_variables: &'a HashMap<String, String>,
+}
+
+impl<'a> TestExecutionSyscallHandler<'a> {
+    pub fn from(
```
Comment (piotmag769) [inline crates/forge/src/test_execution_syscall_handler.rs:79]: `new` imo

File: crates/forge/src/test_execution_syscall_handler.rs [hunk 3]
```diff
@@ -60,15 +60,34 @@ impl From<&StarknetContractArtifacts> for ContractArtifacts {
     }
 }
 
+pub struct TestExecutionState<'a> {
```
Comment (piotmag769) [inline crates/forge/src/test_execution_syscall_handler.rs:63]: I don't like the `State` in name, it suggests mutability and blockchain state. Both of the fields of this struct don't ever change during execution, those are more like `TestExecutionParams` (can use a better name though)

Comment (MaksymilianDemitraszek) [reply crates/forge/src/test_execution_syscall_handler.rs:63]: This is state, which can be mutable if we want. It is for consistency with CheatnetState.

**Review by piotmag769 — APPROVED** [review]

**Pull request merged**